From 7c6854e82994183d0a3ab00d095f33a89957953d Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 13 Nov 2024 13:11:33 +0000
Subject: [PATCH] dt: rp1: Link RP1 DMA to the associated clock

This makes the kernel representation of the clock structure
match reality.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/rp1.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm64/boot/dts/broadcom/rp1.dtsi
@@ -1081,7 +1081,7 @@
 			reg = <0xc0 0x40188000  0x0 0x1000>;
 			compatible = "snps,axi-dma-1.01a";
 			interrupts = <RP1_INT_DMA IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&sdhci_core &rp1_clocks RP1_CLK_SYS>;
+			clocks = <&rp1_clocks RP1_CLK_DMA &rp1_clocks RP1_CLK_SYS>;
 			clock-names = "core-clk", "cfgr-clk";
 
 			#dma-cells = <1>;
